amaranth/vendor/xilinx.py

```diff
@@ -484,16 +578,23 @@ def __init__(self, *, toolchain=None):
             else:
                 toolchain = "Vivado"
 
-        assert toolchain in ("Vivado", "ISE", "Symbiflow")
+        assert toolchain in ("Vivado", "ISE", "Symbiflow", "yosys_nextpnr")
+        self.oss_toolchain = False
```

Comment:
Can you make this a read-only property called `vendor_toolchain`, please? Something like

```python
@property
def vendor_toolchain(self):
    return self.toolchain in ("Vivado", "ISE")
```

Reply:
Donie!